card/lib/card/auth/setup.rb

```diff
@@ -8,10 +8,11 @@ module Setup
       # app is not totally set up yet
       # @return [true/false]
       def needs_setup?
-        @simulating_setup_need || (
-          !Card.cache.read(SETUP_COMPLETED_KEY) &&
-          !Card.cache.write(SETUP_COMPLETED_KEY, account_count > 2)
-        )
+        @simulating_setup_need || begin
+          !Card.cache.fetch(SETUP_COMPLETED_KEY) do
+            account_count > 2
+          end
+        end
```

Comment:
looks much better with `fetch` but I'm not a fan of the additional block. What about
```ruby
@simulating_setup_need ||
  !Card.cache.fetch(SETUP_COMPLETED_KEY) { account > 2 }
```